libdisco/disco.go

```diff
@@ -127,7 +127,7 @@ type handshakeState struct {
 // Serialize is a helper function to serialize a handshake state, later to be unserialized via
 // the `RecoverState()` function.
 // For security purposes, the long-term static keypair is not serialized. Same for the psk
-func (hs handshakeState) Serialize() []byte {
+func (hs *handshakeState) Serialize() []byte {
```

Comment:
this shouldn't be mutable, unless I'm missing something

Reply:
Yep you're missing the consistency in receiver types 
https://github.com/golang/go/wiki/CodeReviewComments#receiver-type

Reply:
Oh interesting, I always thought you shouldn't use a pointer if you don't intent to mutate your value